operator/internal/controller/pulumi/stack_controller.go

```diff
@@ -584,22 +637,22 @@ func (r *StackReconciler) Reconcile(ctx context.Context, request ctrl.Request) (
 		if requireErr != nil {
 			failedPrereqNames = append(failedPrereqNames, prereq.Name)
 			failedPrereqErr = fmt.Errorf("prerequisite not satisfied for %q: %w", prereq.Name, requireErr)
-			// annotate the out of date stack so that it'll be queued. The value is arbitrary; this
-			// value gives a bit of context which might be helpful when troubleshooting.
-			v := fmt.Sprintf("update prerequisite of %s at %s", instance.Name, time.Now().Format(time.RFC3339))
-			prereqStack1 := prereqStack.DeepCopy()
-			a := prereqStack1.GetAnnotations()
-			if a == nil {
-				a = map[string]string{}
-			}
-			a[shared.ReconcileRequestAnnotation] = v
-			prereqStack1.SetAnnotations(a)
-			log.Info("requesting requeue of prerequisite", "name", prereqStack1.Name, "cause", requireErr.Error())
-			if err := r.Client.Patch(ctx, prereqStack1, client.MergeFrom(&prereqStack)); err != nil {
-				// A conflict here may mean the prerequisite has been changed, or it's just been
-				// run. In any case, requeueing this object means we'll see the new state of the
-				// world next time around.
-				return reconcile.Result{}, fmt.Errorf("annotating prerequisite to force requeue: %w", err)
+
+			var errOutOfDate *errRequirementOutOfDate
+			if errors.As(requireErr, &errOutOfDate) {
+				// annotate the out-of-date stack so that it'll be resynced.
+				// The value is arbtrary but here may be understood as the time of the resync that was deemed out-of-date.
+				// Such a value is idempotent (don't want to spam the underlying stack with new requests)
+				// and supports having multiple dependent stacks.
+				setReconcileRequestAnnotation(prereqStack, errOutOfDate.LastResyncTime.Format(time.RFC3339))
```

Comment:
Do we want to no-op if it already had the annotation?

Reply:
If the annotation already has the expected value, then the Update call would effectively be a no-up.